test/storage/replication_test.cpp

```diff
@@ -0,0 +1,239 @@
+#define LOG_FILE_NAME "./test.log"
+
+namespace terrier::storage {
+class ReplicationTests : public TerrierTest {
+ protected:
+  // This is an estimate for how long we expect for all the logs to arrive at the replica and be replayed by the
+  // recovery manager. You should be pessimistic in setting this number, be sure it is an overestimate. Each test should
+  // set it to its own value depending on the amount of work it's doing
+  std::chrono::seconds replication_delay_estimate_;
+
+  // Settings for log manager
+  const uint64_t num_log_buffers_ = 100;
+  const std::chrono::microseconds log_serialization_interval_{10};
+  const std::chrono::milliseconds log_persist_interval_{20};
+  const uint64_t log_persist_threshold_ = (1 << 20);  // 1MB
+  const std::string ip_address_ = "127.0.0.1";
+  const uint16_t replication_port_ = 9022;
+
+  // Settings for server
+  uint32_t max_connections_ = 1;
+  uint32_t conn_backlog_ = 1;
+
+  // Settings for replication
+  const std::chrono::seconds replication_timeout_{10};
+
+  // General settings
+  std::default_random_engine generator_;
+  storage::RecordBufferSegmentPool buffer_pool_{2000, 100};
+  storage::BlockStore block_store_{100, 100};
+
+  // Settings for gc
+  const std::chrono::milliseconds gc_period_{10};
+
+  // Master node's components (prefixed with "master_") in order of initialization
+  // We need:
+  //  1. ThreadRegistry
+  //  2. LogManager
+  //  3. TxnManager
+  //  4. Catalog
+  //  4. GC
+  common::DedicatedThreadRegistry *master_thread_registry_;
+  LogManager *master_log_manager_;
+  transaction::TimestampManager *master_timestamp_manager_;
+  transaction::DeferredActionManager *master_deferred_action_manager_;
+  transaction::TransactionManager *master_txn_manager_;
+  catalog::Catalog *master_catalog_;
+  storage::GarbageCollector *master_gc_;
+  storage::GarbageCollectorThread *master_gc_thread_;
+
+  // Replica node's components (prefixed with "replica_") in order of initialization
+  //  1. Thread Registry
+  //  2. TxnManager
+  //  3. Catalog
+  //  4. GC
+  //  5. RecoveryManager
+  //  6. TrafficCop
+  //  7. TerrierServer
+  common::DedicatedThreadRegistry *replica_thread_registry_;
+  transaction::TimestampManager *replica_timestamp_manager_;
+  transaction::DeferredActionManager *replica_deferred_action_manager_;
+  transaction::TransactionManager *replica_txn_manager_;
+  catalog::Catalog *replica_catalog_;
+  storage::GarbageCollector *replica_gc_;
+  storage::GarbageCollectorThread *replica_gc_thread_;
+  storage::ReplicationLogProvider *replica_log_provider_;
+  storage::RecoveryManager *replica_recovery_manager_;
+  network::ITPCommandFactory *replica_itp_command_factory_;
+  network::ITPProtocolInterpreter::Provider *replica_itp_protocol_provider_;
+  network::ConnectionHandleFactory *replica_connection_handle_factory_;
+  trafficcop::TrafficCop *replica_tcop_;
+  network::TerrierServer *replica_server_;
+
+  void SetUp() override {
+    TerrierTest::SetUp();
```

Comment:
This is an "old style" test prior to DBMain existing, take a look at how tests are brought up now. We don't need to manually create all the components, I think.